packages/plugin-ext/src/plugin/node/debug/plugin-debug-adapter-executable-resolver.ts

```diff
@@ -36,7 +36,10 @@ export async function resolveDebugAdapterExecutable(
         runtime = path.join(pluginPath, runtime);
     }
     const runtimeArgs = info && info.runtimeArgs || debuggerContribution.runtimeArgs || [];
-    const command = runtime ? runtime : program;
+    let command = runtime ? runtime : program;
+    if (command === 'node') {
```

Comment:
I think the correct place to fix is here: https://github.com/veredcon/theia/blob/6167f4b4356c9488a89a86ff7be75ca362122fb7/packages/plugin-ext/src/plugin/node/debug/plugin-debug-adapter-starter.ts#L41-L42

We should check whether `command` is `node` and then use `fork` instead by extracting moduleName from arguments. `fork` ensures that node is inherited.

You don't need to care about electron the plugin host process does not run in Electron already.

Reply:
The thing is that  `startDebugAdapter` expects executable parameter of `theia.DebugAdapterExecutable` which stands only for spawn. In fact I'm not sure today how it enters the `else if ('modulePath' in executable)`?

I would expect `startDebugAdapter`  gets for example `executable: theia.DebugAdapterExecutable | DebugAdapterForkExecutable` and then the change would still be in the same place similar to : https://github.com/eclipse-theia/theia/pull/5508/files#diff-32ee99b4af7468a61598678b4f82086aR234

Reply:
> similar to : https://github.com/eclipse-theia/theia/pull/5508/files#diff-32ee99b4af7468a61598678b4f82086aR234
b

It is not similar where we convert command into modulePath that fork will be used. In this case we cannot do it, so it cannot be fixed here.

`startDebugAdapter` is also good place because it is actually which starts, i.e. the last point where we can figure out that runtime is node and we should fork not spawn.

Reply:
Hi @akosyakov,

I changed the place to be in the `startDebugAdapter`. 
Notice that as also mentioned in my last comment, the code that is checking modulePath was never called so I've deleted it.

I've looked at VSCode and the corresponding code: 
https://github.com/microsoft/vscode/blob/06aadb28a71b09ab3f97a6c0869a8a4a2fa8ba61/src/vs/workbench/contrib/debug/node/debugAdapter.ts#L181

I was not sure how you check whether it is electron so please review this part.